### PR TITLE
Allow testnet ledger accounts

### PR DESCRIPTION
**Pull request by victorkirov**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Enhancement

# 📜 Background
We now have capabilities to run ledger in testnet mode. This enables the creation and management of testnet ledger addresses.

# 🔄 Changes
- Allow creation of Testnet Ledger accounts
- Filter our ledger accounts depending on their network

Impact:
- This will allow us to test Ledger in Testnet without spending real money

# 🖼 Screenshot / 📹 Video

image

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

**Review by DuskaT021 — APPROVED** [review]

**Comment by DuskaT021** [comment]
@victorkirov needs conflicts resolved

**Comment by DuskaT021** [comment]
- missing network option in the settings for switching

image

- issue with ledger and testnet

image

**Comment by victorkirov** [comment]
Sorry @DuskaT021 this wasn't ready for testing 🙈 I made it last night as a placeholder to finish this morning.

I fixed up a bunch of issues and tested it out locally. Should be good for testing now 👍

**Comment by DuskaT021** [comment]
> Sorry @DuskaT021 this wasn't ready for testing 🙈 I made it last night as a placeholder to finish this morning.
> 
> I fixed up a bunch of issues and tested it out locally. Should be good for testing now 👍

the issue with the import ledger on testnet, after the latest fixes is still present: @victorkirov 

image

File: src/app/utils/ledger.ts [hunk 1]
```diff
@@ -26,3 +31,10 @@ export const handleBip322LedgerMessageSigning = async ({
 };
 
 export const signatureVrsToRsv = (sig: string): string => sig.slice(2) + sig.slice(0, 2);
+
+export const filterLedgerAccounts = (accounts: Account[], network: NetworkType) =>
+  accounts.filter(
+    (account) =>
+      (account.ordinalsAddress?.startsWith('bc1') && network === 'Mainnet') ||
```
Comment (dhriaznov) [inline src/app/utils/ledger.ts:38]: Shouldn't this condition be like `account.ordinalsAddress?.startsWith(network === 'Mainnet' ? 'bc1' : 'tb1')` instead?

Comment (victorkirov) [reply src/app/utils/ledger.ts:38]: lol, that makes a lot more sense 👍 Thanks 😄

**Comment by DuskaT021** [comment]
console error for inscribe ordinals on testnet with the ledger account
this endpoint is failing: `https://inscribe-testnet.xverse.app/v1/inscriptions/execute-order`

image

**Comment by victorkirov** [comment]
This is ready again. The issue that came up was that ledger accounts are all in a single list for mainnet and testnet. The device account index used was sequential but it used one sequence for both networks, so, for example,  you'd end up with:
mainnet account - index 0
testnet account - index 1
testnet account - index 2
mainnet account - index 3

Now it separates them when generating an index and we get:
mainnet account - index 0
testnet account - index 0
testnet account - index 1
mainnet account - index 1

**Review by dhriaznov — DISMISSED** [review]

**Comment by DuskaT021** [comment]
@victorkirov we should update the branch :)

**Comment by DuskaT021** [comment]
@m-aboelenein needs conflicts resolved, tagged you cause you have the most context 🙂

**Review by dhriaznov — DISMISSED** [review]

**Comment by DuskaT021** [comment]
@victorkirov branch is outdated

**Review by teebszet — APPROVED** [review]

**Pull request merged**